app/assets/javascripts/discourse/app/services/docking.js

```diff
@@ -0,0 +1,62 @@
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { cancel } from "@ember/runloop";
+import Service from "@ember/service";
+import { disableImplicitInjections } from "discourse/lib/implicit-injections";
+import discourseLater from "discourse-common/lib/later";
+import { debounce } from "discourse-common/utils/decorators";
+
+const INITIAL_DELAY_MS = 50;
+const DEBOUNCE_MS = 5;
+
+@disableImplicitInjections
+export default class Docking extends Service {
+  @tracked dockCheck = null;
+  @tracked _initialTimer = null;
+  @tracked _queuedTimer = null;
+
+  constructor() {
+    super(...arguments);
+
+    window.addEventListener("scroll", this.queueDockCheck, { passive: true });
+    document.addEventListener("touchmove", this.queueDockCheck, {
+      passive: true,
+    });
```

Comment:
Feels kinda weird that we need touchmove for this... any idea why? Feels like it should be entirely handled by scroll position?

I'd be inclined to remove the touchmove listener and see what happens :fire: